test/lightning/attempts_test.exs

```diff
@@ -291,17 +292,77 @@ defmodule Lightning.AttemptsTest do
 
       Lightning.WorkOrders.subscribe(workflow.project_id)
 
+      %{attempt: attempt, workorder_id: workorder.id}
+    end
+
+    test "marks an attempt as started",
+         %{attempt: attempt, workorder_id: workorder_id} do
       assert {:ok, %Attempt{started_at: started_at}} =
                Attempts.start_attempt(attempt)
 
       assert DateTime.compare(started_at, DateTime.utc_now()) == :lt
 
-      workorder_id = workorder.id
-
       assert_received %Lightning.WorkOrders.Events.WorkOrderUpdated{
         work_order: %{id: ^workorder_id}
       }
     end
+
+    test "indicates if a response was unsuccessful", %{attempt: attempt} do
+      with_mock(
+        Lightning.Repo,
+        transaction: fn _multi -> {:error, nil, {:change, :set}, nil} end
```

Comment:
even though it asserts well, would use `%Ecto.Changeset{}` to try to decrease the mocking